CRM/Contact/Form/Edit/Individual.php

```diff
@@ -97,16 +97,16 @@ public static function buildQuickForm($form, $inlineEditMode = NULL): void {
    *   The uploaded files if any.
    * @param int $contactID
    *
-   * @return bool
+   * @return bool|array
    *   TRUE if no errors, else array of errors.
    */
   public static function formRule($fields, $files, $contactID = NULL) {
     $errors = [];
     $primaryID = CRM_Contact_Form_Contact::formRule($fields, $errors, $contactID, 'Individual');
 
     // make sure that firstName and lastName or a primary OpenID is set
-    if (!$primaryID && (empty($fields['first_name']) || empty($fields['last_name']))) {
-      $errors['_qf_default'] = ts('First Name and Last Name OR an email OR an OpenID in the Primary Location should be set.');
+    if (!$primaryID && (empty($fields['first_name']) && empty($fields['last_name']))) {
+      $errors['_qf_default'] = ts('First Name OR Last Name OR an email OR an OpenID in the Primary Location must be set.');
```

Comment:
Can we change the error message to:

> "First Name, Last Name or Email (Primary) must be set."

I'm pretty sure nobody uses OpenID anymore, and the error message is a bit intense. Whenever there are all-caps words, I figure there must be a problem with the sentence (and it does not translate well).

Reply:
Now updated